parcels/grid.py

```diff
@@ -352,6 +371,25 @@ def add_periodic_halo(self, zonal, meridional, halosize=5):
             self.xdim = self.lon.shape[1]
             self.ydim = self.lat.shape[0]
             self.meridional_halo = halosize
+        if isinstance(self, CurvilinearSGrid):
```

Comment:
Since code for `RectilinearSGrid` and `CurvilinearSGrid` is (almost?) the same, create a method?